x-pack/metricbeat/module/azure/monitor_service.go

```diff
@@ -55,16 +55,24 @@ func NewService(clientId string, clientSecret string, tenantId string, subscript
 }
 
 // GetResourceDefinitions will retrieve the azure resources based on the options entered
-func (service MonitorService) GetResourceDefinitions(id []string, group []string, rType string, query string) (resources.ListResultPage, error) {
+func (service MonitorService) GetResourceDefinitions(id []string, group []string, rType string, query string) ([]resources.GenericResource, error) {
 	var resourceQuery string
+	var resourceList []resources.GenericResource
 	if len(id) > 0 {
-		var filterList []string
-		// listing resourceID conditions does not seem to work with the API but querying by name or resource types will work
+		// listing multiple resourceId conditions does not seem to work with the API, extracting the name and resource type does not work as the position of the `resourceType` can move if a parent resource is involved, filtering by resource name and resource group (if extracted) is also not possible as
```

Comment:
Nit. Split this line :slightly_smiling_face:

Reply:
> LGTM, would it be possible to add a test case that would reproduce the original problem?

We will need integration tests in this case, this behavior is based on API supportability, I am hoping to have those in the near future.